pkg/k8s/deleter/deleter.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package deleter
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/d2iq-labs/capi-runtime-extensions/pkg/constants"
+	"github.com/d2iq-labs/capi-runtime-extensions/pkg/k8s/annotations"
+)
+
+type Deleter struct {
+	cluster *v1beta1.Cluster
+	client  client.Client
+	log     logr.Logger
+}
+
+type objectMetaList []metav1.ObjectMeta
+
+func (ol objectMetaList) asCommaSeparatedString() string {
+	out := ""
+	separator := ""
+	for n := range ol {
+		object := ol[n]
+		out += fmt.Sprintf("%s%s/%s", separator, object.Namespace, object.Name)
+		separator = ", "
+	}
+	return out
+}
+
+func New(log logr.Logger, cluster *v1beta1.Cluster, remoteClient client.Client) Deleter {
+	return Deleter{
+		cluster: cluster,
+		client:  remoteClient,
+		log:     log,
+	}
+}
+
+func (d *Deleter) DeleteServicesWithLoadBalancer(ctx context.Context) error {
+	err := deleteServicesWithLoadBalancer(ctx, d.client, d.log)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func deleteServicesWithLoadBalancer(
+	ctx context.Context,
+	c client.Client,
+	log logr.Logger,
+) error {
+	log.Info("Listing Services with type LoadBalancer")
+	services := &corev1.ServiceList{}
+	err := c.List(ctx, services)
+	if err != nil {
+		return fmt.Errorf("error listing Services: %w", err)
+	}
+
+	svcsFailedToBeDeleted := make(objectMetaList, 0)
```

Comment:
nit: Better to just nil assign this rather than create an empty list to avoid potentially unnecessary allocation.
```suggestion
	var svcsFailedToBeDeleted objectMetaList
```